src/student_success_tool/analysis/pdp/targets/shared.py

```diff
@@ -281,6 +283,55 @@ def _compute_intensity_num_terms(
     ]
 
 
+def get_nth_student_terms(
+    df: pd.DataFrame,
+    *,
+    student_id_cols: str | list[str] = "student_guid",
+    n: int = 1,
```

Comment:
I recommend moving `n` up above the cols args, and _not_ giving it a default value -- i.e. making it a required arg. I feel pretty strongly that choosing `n` should always be explicit.